cmd/deployment/metadata.go

```diff
@@ -0,0 +1,72 @@
+package project_map
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+	errors "github.com/pkg/errors"
+	"github.com/urfave/cli/v2"
+)
+
+type Metadata struct {
+	Deployment Item `json:"deployment"`
+}
+
+type Item struct {
+	Sha        string `json:"sha"`
+	AuthorName string `json:"authorName"`
+	When       string `json:"when"`
+	Tag        string `json:"tag"`
+	Msg        string `json:"msg"`
+}
+
+func DeploymentMetadata(c *cli.Context) error {
+	repo, err := git.PlainOpenWithOptions(".", &git.PlainOpenOptions{DetectDotGit: true})
+	if err != nil {
+		return errors.Wrap(err, "unable to open git repository")
+	}
+
+	ref, err := repo.Head()
+	if err != nil {
+		return errors.Wrap(err, "unable to get HEAD reference")
+	}
+	fmt.Println(ref.Hash())
+
+	msg, err := repo.CommitObject(ref.Hash())
+	if err != nil {
+		return errors.Wrap(err, "unable to get commit object")
+	}
+
+	tagString := ""
```

Comment:
if no tag is present that value of `tag` on [line 44](https://github.com/dpc-sdp/bay-cli/pull/27/files#diff-1262dd46af24983711b1ba1a2507ed3ab257004339bbcddfdc43871071ce572eR44) is `<nil>` which I couldn't output as a string. This handling to get a string works but is it brittle or obsfucate errors by hijacking the err on [line 47](https://github.com/dpc-sdp/bay-cli/pull/27/files#diff-1262dd46af24983711b1ba1a2507ed3ab257004339bbcddfdc43871071ce572eR47)?